### PR TITLE
Add CI flag -cilium.multinode[=true]

### PR DESCRIPTION
**Pull request by joestringer**
I've previously had some success running the `K8sDatapathConfig` tests on a local microk8s (single-node) cluster by hacking some variables in the tests. Recent refactoring as part of the manifest deploymentmanager has broken this workflow, so I figured it's about time to formalize this to support this use case properly.

This PR adds a new ginkgo flag, `-cilium.multinode` which defaults to true (the current behaviour); if set to false, or if running with `CNI_INTEGRATION={microk8s,minikube}` , it allows running tests that assume multinode connectivity within a single node by avoiding nodeselectors that require a second node. This can be useful for some basic debugging of the tests in a single-node environment before moving to a more complicated environment. Note that the node must still have the k8s1 ci node label.

### CODE REVIEW

**Comment by joestringer** [comment]
test-me-please

**Review by christarazi — COMMENTED** [review]
Fancy! Can you add this to the docs?

**Comment by joestringer** [comment]
Hit #11560 in Travis tests.

**Comment by joestringer** [comment]
@christarazi sure. I've fixed up the docs and posted a PR for that. Will wait for this to merge before merging that since it refers to the flag proposed in this PR. https://github.com/cilium/cilium/pull/11629

**Review by christarazi — APPROVED** [review]

**Review by michi-covalent — APPROVED** [review]

**Comment by joestringer** [comment]
Hit #6728 in `K8s-1.18-kernel-4.9` run. Known (occasional) flake.

File: test/k8sT/manifests/demo_ds_local.yaml
Comment (joestringer) [inline test/k8sT/manifests/demo_ds_local.yaml:94]: Note that the only diff with the standard `demo_ds.yaml` here is the `nodeSelector`. If we had better templating, this extra file wouldn't be necessary but I figured this is the easiest way to get this working right now.

**Comment by joestringer** [comment]
No additional feedback from CI folks during EU hours, so doesn't seem like there's anything too objectionable here. It doesn't break overall CI and it was working for me locally. We can follow up with any subsequent feedback. Thanks for the reviews, merging.

**Pull request merged**
